tests/MongoDB.Driver.Core.Tests/Core/Configuration/ClusterBuilderTests.cs

```diff
@@ -59,10 +63,56 @@ public void CreateServerMonitorFactory_should_return_expected_result(int connect
 
             var serverSettings = result._settings();
         }
+
+        [Fact]
+        public void RegisterStreamFactory_should_complement_previous_RegisterStreamFactory_call_in_right_order()
+        {
+            var subject = new ClusterBuilder();
+            subject.RegisterStreamFactory(
+                factory => {
+                    var testFactory = (StreamFactoryTest)factory;
+                    testFactory.TestValue.Should().Be(1);
+                    testFactory.TestValue++;
+                    return factory;
+                });
+            subject.RegisterStreamFactory(
+                factory => {
+                    var testFactory = (StreamFactoryTest)factory;
+                    testFactory.TestValue.Should().Be(2);
+                    testFactory.TestValue++;
+                    return factory;
+                });
+            subject.RegisterStreamFactory(
+                factory => {
+                    var testFactory = (StreamFactoryTest)factory;
+                    testFactory.TestValue.Should().Be(3);
+                    testFactory.TestValue++;
+                    return factory;
+                });
+            var streamFactoryWrapper = subject._streamFactoryWrapper();
+            streamFactoryWrapper(new StreamFactoryTest { TestValue = 1 });
+        }
+    }
```

Comment:
While trying to figure out how this test works it occurred to me that the test could be simpler:

    [Fact]
    public void StreamFactoryWrappers_should_be_called_in_the_correct_order()
    {
        var subject = new ClusterBuilder();
        var calls = new List<int>();
        subject.RegisterStreamFactory(factory => { calls.Add(1); return factory; });
        subject.RegisterStreamFactory(factory => { calls.Add(2); return factory; });
        subject.RegisterStreamFactory(factory => { calls.Add(3); return factory; });

        subject._streamFactoryWrapper()(Mock.Of<IStreamFactory>());

        calls.Should().Equal(new[] { 1, 2, 3 });
    }

And not even require a FakeStreamFactory.

Note that the general pattern for blank lines in tests is one blank line between the Arrange/Act/Assert sections. This pattern can be broken for very complicated tests.

Reply:
Ok, I will update the code in this way.